src/db/sysdb_ops.c

```diff
@@ -3972,6 +4014,24 @@ int sysdb_delete_user(struct sss_domain_info *domain,
             }
         }
 
+        /* If user has a linked userOverride delete it */
+        ret = sysdb_user_local_override_dn(tmp_ctx, domain, msg->dn,
+                                           &override_dn);
+        if (ret != EOK) {
+            DEBUG(SSSDBG_OP_FAILURE,
+                  "Failed to build local override DN: %s\n",
+                  strerror(ret));
+            goto fail;
+        }
+
```

Comment:
I think it would be sufficient to just cover the 2 deletes in the transaction. Since we do not have multiple threads there is no race where a different thread can delete the entries while this code would be between the search and the deletes.

Reply:
But you could get an error inside the for loop and the ghost user would end removed from a partial set of groups, isn't?

Reply:
That's a fair point. But even here I would prefer to have the transaction only wrapping the for-loop to lock the database as short as possible. On the other hand one might argue that if the user should be deleted, deleting as much ghosts entries as possible id better than none and it would be better to use no transaction for the ghost entries at all.

Since this is not strictly related to overrides, feel free to open a ticket.